airflow-core/src/airflow/serialization/serialized_objects.py

```diff
@@ -2172,10 +2172,10 @@ def get_run_data_interval(self, run: DagRun) -> DataInterval:
             raise ValueError(f"Arguments refer to different DAGs: {self.dag_id} != {run.dag_id}")
 
         data_interval = _get_model_data_interval(run, "data_interval_start", "data_interval_end")
-        # the older implementation has call to infer_automated_data_interval if data_interval is None, do we want to keep that or raise
-        # an exception?
         if data_interval is None:
-            raise ValueError(f"Cannot calculate data interval for run {run}")
+            if run.logical_date is None:
+                raise ValueError(f"Cannot calculate data interval for run {run}")
```

Comment:
If logical_date is None, we should simply allow the data interval to be None without error.

Reply:
Got it, I thought it was forbidden. Just updated it